tools/grammars/compiler/converter.go

```diff
@@ -65,6 +65,9 @@ func (conv *Converter) tmpScopes() map[string]bool {
 func (conv *Converter) AddGrammar(source string) error {
 	repo := conv.Load(source)
 	if len(repo.Files) == 0 {
+		if len(repo.Errors) > 0 {
+			return fmt.Errorf("Parsing error in '%s': %s", source, repo.Errors[0])
+		}
```

Comment:
This feels a bit odd to me: "We found no grammar files, but hey, here's an error."

I'm mostly AFK this week so don't have time to dig into this deeper myself, but couldn't we check for errors in the first `if`, eg:

```golang
if len(repo.Files) == 0 && len(repo.Errors) == 0 {
	return fmt.Errorf("source '%s' contains no grammar files", source)
}
```

... and only return if we really don't find any files and leave the error reporting for the check slightly further down?

Reply:
Oh, right! That should work.

I'll update the pull request tonight.

Reply:
Done.